python/tvm/relay/quantize/quantize.py

```diff
@@ -373,3 +716,135 @@ def quantize(graph, params=None, dataset=None):
     graph = realize(graph)
     graph = _ir_pass.fold_constant(graph)
     return graph
+
+def _evaluate(val_data, batch_fn, graph, lib, params, ctx, free_vars=[], config=[], num_classes=1000, early_stopping=32, log_iter=2):
+    import mxnet as mx
+    """Evaluate function for profiling."""
+    import tvm
+    import logging
+    logging.basicConfig(level=logging.INFO)
+    from tvm.contrib import graph_runtime
+
+    # create runtime module
+    m = graph_runtime.create(graph, lib, ctx)
+    scales = {}
+
+    for i in range(0, len(free_vars)):
+        free_var = free_vars[i]
+        if i >= len(config):
+            shape = m.get_input(i+1).shape
+            dummy = np.empty(shape=shape)
+            if len(dummy.shape) > 0:
+                dummy[:] = np.nan
+            else:
+                dummy = np.nan
+            params[str(free_var.name_hint)] = np.array(dummy)
+        else:
+            params[str(free_var.name_hint)] = np.array(config[i]/128)
+
+    m.set_input(**params)
+    batch_size = 1
+    oshape = (batch_size, num_classes)
+    out_arr = tvm.nd.empty(oshape, "float32")
+    # setup evaluaiton metric
+    acc_top1 = mx.metric.Accuracy()
+    acc_top5 = mx.metric.TopKAccuracy(5)
+    val_data.reset()
```

Comment:
We should not assume calibration data comes from MXNET iterator. I think a Python list of tuple is enough, as we only need around thousand images for calibration.

Reply:
In this case we only need ~32 images for calibration, but I agree that passing NDArrays can work.